core/src/main/scala/org/apache/spark/util/Clock.scala

```diff
@@ -17,11 +17,21 @@
 
 package org.apache.spark.util
 
+import java.util.concurrent.TimeUnit
+
 /**
  * An interface to represent clocks, so that they can be mocked out in unit tests.
  */
 private[spark] trait Clock {
+  /** @return the time in milliseconds */
   def getTimeMillis(): Long
+
+  /**
+   * Wait until a clock's view of current time reaches the specified target time.
+   *
+   * @param targetTime block until the current time is at least this value
+   * @return clock's view of current time when the wait has completed
+   */
   def waitTillTime(targetTime: Long): Long
```

Comment:
it'd be good to rename the variable to reflect the unit, similar to getTimeMillis